src/Panel/Panel.php

```diff
@@ -471,7 +472,19 @@ public static function routesForViews(string $areaId, array $area): array
 		foreach ($views as $view) {
 			$view['area'] = $areaId;
 			$view['type'] = 'view';
-			$routes[] = $view;
+
+			$when = $view['when'] ?? null;
+
+			// checks if there is a load view condition
+			if ($when instanceof Closure) {
+				if ($when($view, $area) === true) {
+					$routes[] = $view;
+				}
+			} else {
+				$routes[] = $view;
+			}
+
+			unset($view['when']);
```

Comment:
This won't take effect on the `$routes` array as the `$view` array is copied to `$routes` by value, not by reference.

We could solve this by adding `$key => $view` in line 472, using that `$key` in lines 481 and 484 and then doing this here:

```suggestion
			unset($routes[$key]['when']);
```

Reply:
Correct. But I've fixed with another simple way 🙂